tests/simple/common/shell_tests.sh

```diff
@@ -0,0 +1,13 @@
+test_shell_syntax() {
+	bash -n "${1}"
+}
+
+test_shellcheck() {
+	if ! which shellcheck ; then
+		echo "ShellCheck not found: skipping..."
+		return 0
+	fi
+
+	shellcheck -x -s bash -e SC2181,SC2029,SC1091,SC1090 "${1}"
```

Comment:
Yes, brackets denote subshells and bash processes all contents within as raw bash (so you don't have to e.g. escape quotes) which also means it has to be valid/shellcheck-correct syntax on its own.